packages/flutter_tools/lib/src/isolated/native_assets/native_assets.dart

```diff
@@ -84,21 +143,25 @@ Future<DartBuildResult> runFlutterSpecificDartBuild({
     await buildDir.create(recursive: true);
   }
 
-  if (!await _nativeBuildRequired(buildRunner)) {
-    return const DartBuildResult.empty();
+  if (!await _hookRunRequired(buildRunner)) {
+    return DartHookResult.empty();
   }
 
   final BuildMode buildMode = _getBuildMode(environmentDefines, flutterTester);
-  final List<Architecture> architectures =
-      flutterTester
-          ? <Architecture>[Architecture.current]
-          : _architecturesForOS(targetPlatform, targetOS, environmentDefines);
-  final DartBuildResult result =
-      architectures.isEmpty
-          ? const DartBuildResult.empty()
-          : await _runDartBuild(
+  final List<Architecture>? architectures =
```

Comment:
Should we consider adding a `Architecture.web`, `Architecture.nonNative`, or `Architecture.unsupported` to `package:native_assets_cli`? I just wonder if there's a cleaner way to handle this that makes it more obvious what's going on.

Reply:
Hm, sounds like a good idea to me, at least specifying the web case a bit more. Let me open a PR on dart-lang/native.

Reply:
(Resolving as I added some todos around this)